packages/server/src/main/java/software/uncharted/terarium/hmiserver/service/data/TerariumAssetService.java

```diff
@@ -158,7 +159,7 @@ public T createAsset(final T asset) throws IOException {
 	 */
 	@Override
 	@Observed(name = "function_profile")
-	public List<T> createAssets(final List<T> assets) throws IOException {
+	public List<T> createAssets(final List<T> assets, final Schema.Permission hasWritePermission) throws IOException {
```

Comment:
I don't get it here? Why passing that `hasWritePermission` all the way from `Controller -> Xservice -> ITerariumAssetService -> TerariumAssetService` to not use it?
If the user cannot create an asset on this project, shouldn't be the `Controller` to return an 403 error?

Reply:
It is the only way I can think of that informs the developer that they need to check for permissions on these calls, the fact that it is an unused parameter lower down is a "feature" of getting the compiler to inform the developer they have missed something vital and totally tangential.  A comment would be absolutely ignored.

Reply:
I suppose I could add checks to satisfy the parameter, but they are checking for the non-absence of a permission at the wrong spot and as you say, there is not the right place to throw a 403.  Instead the `Controller` (as I have written the implementation) is first determining if the user even has the correct rights to continue before doing any additional work on the server.